classes/Hook/HookDisplayOrderConfirmation.php

```diff
@@ -49,60 +49,58 @@ public function __construct(Ps_Googleanalytics $module, Context $context)
      */
     public function run()
     {
+        $gaScripts = '';
         $order = $this->params['order'];
 
-        if (Validate::isLoadedObject($order) && $order->getCurrentState() != (int) Configuration::get('PS_OS_ERROR')) {
-            $ganalyticsRepository = new GanalyticsRepository();
-            $gaOrderSent = $ganalyticsRepository->findGaOrderByOrderId((int) $order->id);
+        if (!Validate::isLoadedObject($order) || $order->getCurrentState() == (int) Configuration::get('PS_OS_ERROR')) {
+            return $gaScripts;
+        }
 
-            if (false === $gaOrderSent) {
-                $ganalyticsRepository->addNewRow(
-                    [
-                        'id_order' => (int) $order->id,
-                        'id_shop' => (int) $this->context->shop->id,
-                        'sent' => 0,
-                        'date_add' => ['value' => 'NOW()', 'type' => 'sql'],
-                    ]
-                );
+        // Load up our handlers and repositories
+        $ganalyticsRepository = new GanalyticsRepository();
+        $gaTagHandler = new GanalyticsJsHandler($this->module, $this->context);
```

Comment:
You're loading services by yourself. Cannot we use (Symfony DI](https://symfony.com/doc/current/components/dependency_injection.html) instead ?

It will do the job nice and once for you ;)